lib/streamlit/connections/snowpark_connection.py

```diff
@@ -0,0 +1,158 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
```

Comment:
@sfc-gh-yixie heads up that we have a first pass at a Snowpark connection class here in case you have any thoughts on it. (It may also be useful to skim the `ExperimentalBaseConnection` implementation in `lib/streamlit/connections/base_connection.py`)

We plan on releasing this soon as an experimental/PuPr feature, so we do have some leeway in being able to make changes to the connection class API in the future if necessary